### DISCUSSION TITLE
Error message while installing sbase in CI

### DISCUSSION THREAD

**Discussion by maburrub** [seleniumbase/SeleniumBase#1124] [Q&A] [answered]
I created a python virtual env then installed sbase using:
pip install seleniumbase
then I did pip freeze > requirements.txt
now I can install these dependences any venv of my choice by
pip install -r requiremets.txt
This works fine on my local mac machine but when I apply the same steps on a gitlab linux runner. I get the following error:
Note: the error is not preventing seleniumbase from working, everything is working fine but I am wondering what is this error message. 

Collecting wheel>=0.37.0
  Downloading wheel-0.37.0-py2.py3-none-any.whl (35 kB)
Building wheels for collected packages: ipdb
  Building wheel for ipdb (setup.py): started
  Building wheel for ipdb (setup.py): finished with status 'error'
  ERROR: Command errored out with exit status 1:
   command: /builds/mabualrob/pacautomation/env/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-68y7am09/ipdb/setup.py'"'"'; __file__='"'"'/tmp/pip-install-68y7am09/ipdb/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-zvre04so
       cwd: /tmp/pip-install-68y7am09/ipdb/
  Complete output (6 lines):
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help
  
  error: invalid command 'bdist_wheel'
  ----------------------------------------
  ERROR: Failed building wheel for ipdb
  Running setup.py clean for ipdb
Failed to build ipdb
ERROR: seleniumbase 2.2.10 has requirement setuptools>=59.6.0; python_version >= "3.6", but you'll have setuptools 44.0.0 which is incompatible.

**Comment by mdmintz** [comment, marked as answer]
@maburrub Your version of ``setuptools`` is so old, (``44.0.0`` vs latest ``59.6.0``), that you'll probably have to upgrade that first, independently, before installing ``seleniumbase``:
``pip install --upgrade setuptools``
And you should do the same for ``pip`` too:
``pip install --upgrade pip``
Then you can install the rest as needed.
